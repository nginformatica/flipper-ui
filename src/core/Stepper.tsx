import { Step, StepLabel, Stepper as MuiStepper } from '@material-ui/core'
import React, { Component } from 'react'
import { DefaultProps } from './types'

export interface StepperProps extends DefaultProps {
    active?: number
    steps: Array<string | TStep>
    bottomLabel?: boolean
    orientation?: 'horizontal' | 'vertical'
}

type TStep = {
    label: JSX.Element | string
    icon: JSX.Element | ((active?: boolean) => JSX.Element)
}

const isActive = (index: number, active: StepperProps['active']) =>
    active !== undefined ? active >= index : undefined

class Stepper extends Component<StepperProps, {}> {
    public static defaultProps = { active: 0 }

    public render() {
        const {
            active,
            bottomLabel,
            steps,
            padding,
            margin,
            style = {},
            ...otherProps
        } = this.props

        return (
            <MuiStepper
                alternativeLabel={bottomLabel}
                activeStep={active}
                style={{ padding, margin, ...style }}
                {...otherProps}>
                {steps.map((step, index) => (
                    <Step key={index}>
                        <StepLabel
                            icon={
                                typeof step === 'object'
                                    ? typeof step.icon === 'function'
                                        ? step.icon(isActive(index, active))
                                        : React.cloneElement(step.icon, {
                                              active: isActive(index, active),
                                              color: isActive(index, active)
                                                  ? 'primary'
                                                  : 'default'
                                          })
                                    : index + 1
                            }>
                            {typeof step === 'object' ? step.label : step}
                        </StepLabel>
                    </Step>
                ))}
            </MuiStepper>
        )
    }
}

export default Stepper
