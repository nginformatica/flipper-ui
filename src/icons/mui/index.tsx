export { default as IconAccessible } from '@mui/icons-material/Accessible'
export { default as IconAdd } from '@mui/icons-material/Add'
export { default as IconAccountCircle } from '@mui/icons-material/AccountCircle'
export { default as IconAddCircleOutline } from '@mui/icons-material/AddCircleOutline'
export { default as IconAllInbox } from '@mui/icons-material/AllInbox'
export { default as IconArrowBack } from '@mui/icons-material/ArrowBack'
export { default as IconArrowBackIos } from '@mui/icons-material/ArrowBackIos'
export { default as IconAssignment } from '@mui/icons-material/Assignment'
export { default as IconAssignmentInd } from '@mui/icons-material/AssignmentInd'
export { default as IconAudiotrack } from '@mui/icons-material/Audiotrack'
export { default as IconArrowDownward } from '@mui/icons-material/ArrowDownward'
export { default as IconArrowDropDown } from '@mui/icons-material/ArrowDropDown'
export { default as IconArrowUpward } from '@mui/icons-material/ArrowUpward'
export { default as IconBackHand } from '@mui/icons-material/BackHand'
export { default as IconBallot } from '@mui/icons-material/Ballot'
export { default as IconBiotechSharp } from '@mui/icons-material/BiotechSharp'
export { default as IconBlock } from '@mui/icons-material/Block'
export { default as IconBorderColor } from '@mui/icons-material/BorderColor'
export { default as IconBuild } from '@mui/icons-material/Build'
export { default as IconBusiness } from '@mui/icons-material/Business'
export { default as IconCancelOutlined } from '@mui/icons-material/CancelOutlined'
export { default as IconCached } from '@mui/icons-material/Cached'
export { default as IconCallReceived } from '@mui/icons-material/CallReceived'
export { default as IconCheck } from '@mui/icons-material/Check'
export { default as IconCheckBox } from '@mui/icons-material/CheckBox'
export { default as IconCheckCircle } from '@mui/icons-material/CheckCircle'
export { default as IconCheckCircleOutlined } from '@mui/icons-material/CheckCircleOutlined'
export { default as IconCircle } from '@mui/icons-material/Circle'
export { default as IconClose } from '@mui/icons-material/Close'
export { default as IconCloudDownload } from '@mui/icons-material/CloudDownload'
export { default as IconCompareArrows } from '@mui/icons-material/CompareArrows'
export { default as IconContentCopy } from '@mui/icons-material/ContentCopy'
export { default as IconContentPaste } from '@mui/icons-material/ContentPaste'
export { default as IconCreditCard } from '@mui/icons-material/CreditCard'
export { default as IconDateRange } from '@mui/icons-material/DateRange'
export { default as IconDelete } from '@mui/icons-material/Delete'
export { default as IconDescription } from '@mui/icons-material/Description'
export { default as IconDescriptionOutlined } from '@mui/icons-material/DescriptionOutlined'
export { default as IconDoneAll } from '@mui/icons-material/DoneAll'
export { default as IconDonutLarge } from '@mui/icons-material/DonutLarge'
export { default as IconDragHandle } from '@mui/icons-material/DragHandle'
export { default as IconDragIndicator } from '@mui/icons-material/DragIndicator'
export { default as IconEdit } from '@mui/icons-material/Edit'
export { default as IconEmail } from '@mui/icons-material/Email'
export { default as IconEnhancedEncryption } from '@mui/icons-material/EnhancedEncryption'
export { default as IconEqualizer } from '@mui/icons-material/Equalizer'
export { default as IconError } from '@mui/icons-material/Error'
export { default as IconErrorOutline } from '@mui/icons-material/ErrorOutline'
export { default as IconExitToApp } from '@mui/icons-material/ExitToApp'
export { default as IconExpand } from '@mui/icons-material/Expand'
export { default as IconExpandLess } from '@mui/icons-material/ExpandLess'
export { default as IconExpandMore } from '@mui/icons-material/ExpandMore'
export { default as IconFace } from '@mui/icons-material/Face'
export { default as IconFactCheck } from '@mui/icons-material/FactCheck'
export { default as IconFavorite } from '@mui/icons-material/Favorite'
export { default as IconFeedbackOutlined } from '@mui/icons-material/FeedbackOutlined'
export { default as IconFileCopy } from '@mui/icons-material/FileCopy'
export { default as IconFilterList } from '@mui/icons-material/FilterList'
export { default as IconFirstPage } from '@mui/icons-material/FirstPage'
export { default as IconFolderShared } from '@mui/icons-material/FolderShared'
export { default as IconFormatColorFill } from '@mui/icons-material/FormatColorFill'
export { default as IconFormatColorText } from '@mui/icons-material/FormatColorText'
export { default as IconFormatListBulleted } from '@mui/icons-material/FormatListBulleted'
export { default as IconForumOutlined } from '@mui/icons-material/ForumOutlined'
export { default as IconFullscreenRounded } from '@mui/icons-material/FullscreenRounded'
export { default as IconHeadset } from '@mui/icons-material/Headset'
export { default as IconHelp } from '@mui/icons-material/Help'
export { default as IconHelpOutline } from '@mui/icons-material/HelpOutline'
export { default as IconHistory } from '@mui/icons-material/History'
export { default as IconHome } from '@mui/icons-material/Home'
export { default as IconHomeWorkOutlined } from '@mui/icons-material/HomeWorkOutlined'
export { default as IconHourglassFull } from '@mui/icons-material/HourglassFull'
export { default as IconVote } from '@mui/icons-material/HowToVote'
export { default as IconImportantDevices } from '@mui/icons-material/ImportantDevices'
export { default as IconInfo } from '@mui/icons-material/Info'
export { default as IconInsertDriveFile } from '@mui/icons-material/InsertDriveFile'
export { default as IconLabelImportant } from '@mui/icons-material/LabelImportant'
export { default as IconLastPage } from '@mui/icons-material/LastPage'
export { default as IconLaunch } from '@mui/icons-material/Launch'
export { default as IconLibraryBooksOutlined } from '@mui/icons-material/LibraryBooksOutlined'
export { default as IconList } from '@mui/icons-material/List'
export { default as IconListAlt } from '@mui/icons-material/ListAlt'
export { default as IconLiveHelp } from '@mui/icons-material/LiveHelp'
export { default as IconLocalOffer } from '@mui/icons-material/LocalOffer'
export { default as IconLock } from '@mui/icons-material/Lock'
export { default as IconLockOpen } from '@mui/icons-material/LockOpen'
export { default as IconLockOutlined } from '@mui/icons-material/LockOutlined'
export { default as IconLocalHospital } from '@mui/icons-material/LocalHospital'
export { default as IconMenu } from '@mui/icons-material/Menu'
export { default as IconMonitorHeart } from '@mui/icons-material/MonitorHeart'
export { default as IconMoreHoriz } from '@mui/icons-material/MoreHoriz'
export { default as IconMoreVert } from '@mui/icons-material/MoreVert'
export { default as IconMovie } from '@mui/icons-material/Movie'
export { default as IconNavigateBefore } from '@mui/icons-material/NavigateBefore'
export { default as IconNavigateNext } from '@mui/icons-material/NavigateNext'
export { default as IconNoteAdd } from '@mui/icons-material/NoteAdd'
export { default as IconNotifications } from '@mui/icons-material/Notifications'
export { default as IconOndemandVideoOutlined } from '@mui/icons-material/OndemandVideoOutlined'
export { default as IconPeopleAlt } from '@mui/icons-material/PeopleAlt'
export { default as IconPerson } from '@mui/icons-material/Person'
export { default as IconPersonAdd } from '@mui/icons-material/PersonAdd'
export { default as IconPersonPin } from '@mui/icons-material/PersonPin'
export { default as IconPhoto } from '@mui/icons-material/Photo'
export { default as IconPhotoCamera } from '@mui/icons-material/PhotoCamera'
export { default as IconPin } from '@mui/icons-material/Pin'
export { default as IconPlayCircleFilled } from '@mui/icons-material/PlayCircleFilled'
export { default as IconPrint } from '@mui/icons-material/Print'
export { default as IconPriorityHigh } from '@mui/icons-material/PriorityHigh'
export { default as IconRecentActors } from '@mui/icons-material/RecentActors'
export { default as IconRemoveCircleOutline } from '@mui/icons-material/RemoveCircleOutline'
export { default as IconRepeat } from '@mui/icons-material/Repeat'
export { default as IconReplay } from '@mui/icons-material/Replay'
export { default as IconReportOutlined } from '@mui/icons-material/ReportOutlined'
export { default as IconRestore } from '@mui/icons-material/Restore'
export { default as IconRoom } from '@mui/icons-material/Room'
export { default as IconRoomOutlined } from '@mui/icons-material/RoomOutlined'
export { default as IconSave } from '@mui/icons-material/Save'
export { default as IconSaveAlt } from '@mui/icons-material/SaveAlt'
export { default as IconSchedule } from '@mui/icons-material/Schedule'
export { default as IconSearch } from '@mui/icons-material/Search'
export { default as IconSentimentNeutral } from '@mui/icons-material/SentimentNeutral'
export { default as IconSentimentSatisfied } from '@mui/icons-material/SentimentSatisfied'
export { default as IconSentimentSatisfiedAlt } from '@mui/icons-material/SentimentSatisfiedAlt'
export { default as IconSentimentDissatisfied } from '@mui/icons-material/SentimentDissatisfied'
export { default as IconSentimentVeryDissatisfied } from '@mui/icons-material/SentimentVeryDissatisfied'
export { default as IconSentimentVerySatisfied } from '@mui/icons-material/SentimentVerySatisfied'
export { default as IconSensors } from '@mui/icons-material/Sensors'
export { default as IconSettings } from '@mui/icons-material/Settings'
export { default as IconSort } from '@mui/icons-material/Sort'
export { default as IconStar } from '@mui/icons-material/Star'
export { default as IconStarBorder } from '@mui/icons-material/StarBorder'
export { default as IconSync } from '@mui/icons-material/Sync'
export { default as IconSyncProblem } from '@mui/icons-material/SyncProblem'
export { default as IconTextFields } from '@mui/icons-material/TextFields'
export { default as IconTimeline } from '@mui/icons-material/Timeline'
export { default as IconToday } from '@mui/icons-material/Today'
export { default as IconUpdate } from '@mui/icons-material/Update'
export { default as IconVisibility } from '@mui/icons-material/Visibility'
export { default as IconVisibilityOff } from '@mui/icons-material/VisibilityOff'
export { default as IconVpnKey } from '@mui/icons-material/VpnKey'
export { default as IconWarning } from '@mui/icons-material/Warning'
export { default as IconWaterDrop } from '@mui/icons-material/WaterDrop'
export { default as IconWifiOff } from '@mui/icons-material/WifiOff'
