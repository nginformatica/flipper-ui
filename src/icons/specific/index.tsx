export { IconAccident } from './IconAccident'
export { IconActionPlans } from './IconActionPlans'
export { IconAddDocument } from './IconAddDocument'
export { IconAET } from './IconAET'
export { IconAnwser } from './IconAnwser'
export { IconASO } from './IconASO'
export { IconBelt } from './IconBelt'
export { IconBillet } from './IconBillet'
export { IconBiometry } from './IconBiometry'
export { IconBoot } from './IconBoot'
export { IconBranch } from './IconBranch'
export { IconBrigades } from './IconBrigades'
export { IconCalendarDay } from './IconCalendarDay'
export { IconCalendarMonth } from './IconCalendarMonth'
export { IconCalendarWeek } from './IconCalendarWeek'
export { IconCancelRounded } from './IconCancelRounded'
export { IconCards } from './IconCards'
export { IconChangeEquipment } from './IconChangeEquipment'
export { IconCircle } from './IconCircle'
export { IconCompany } from './IconCompany'
export { IconComponent } from './IconComponent'
export { IconConfirmRounded } from './IconConfirmRounded'
export { IconCostCenter } from './IconCostCenter'
export { IconCoverall } from './IconCoverall'
export { IconDashboard } from './IconDashboard'
export { IconDistribution } from './IconDistribution'
export { IconDoctor } from './IconDoctor'
export { IconDoctorCalendar } from './IconDoctorCalendar'
export { IconDonator } from './IconDonator'
export { IconDoneChecked } from './IconDoneChecked'
export { IconDoneUnchecked } from './IconDoneUnchecked'
export { IconDownloadQRCode } from './IconDownloadQRCode'
export { IconEmployee } from './IconEmployee'
export { IconEnvironmentalReports } from './IconEnvironmentalReports'
export { IconEquipment } from './IconEquipment'
export { IconErrorStroke } from './IconErrorStroke'
export { IconESocial } from './IconESocial'
export { IconExtinguishingSystem } from './IconExtinguishingSystem'
export { IconFaceScan } from './IconFaceScan'
export { IconFemale } from './IconFemale'
export { IconFileInform } from './IconFileInform'
export { IconFluig } from './IconFluig'
export { IconFunil } from './IconFunil'
export { IconGeneral } from './IconGeneral'
export { IconGift } from './IconGift'
export { IconHealth } from './IconHealth'
export { IconHeight } from './IconHeight'
export { IconIMC } from './IconIMC'
export { IconIPE } from './IconIPE'
export { IconLocalization } from './IconLocalization'
export { IconLTCAT } from './IconLTCAT'
export { IconLTP } from './IconLTP'
export { IconMail } from './IconMail'
export { IconMale } from './IconMale'
export { IconMandates } from './IconMandates'
export { IconMap } from './IconMap'
export { IconMask } from './IconMask'
export { IconMaterial } from './IconMaterial'
export { IconMedicalRecords } from './IconMedicalRecords'
export { IconMediumCompany } from './IconMediumCompany'
export { IconMegaCompany } from './IconMegaCompany'
export { IconMinimumCompany } from './IconMinimumCompany'
export { IconNickName } from './IconNickName'
export { IconOccupation } from './IconOccupation'
export { IconPAE } from './IconPAE'
export { IconPCMAT } from './IconPCMAT'
export { IconPCMSO } from './IconPCMSO'
export { IconPGR } from './IconPGR'
export { IconPGRTR } from './IconPGRTR'
export { IconPPR } from './IconPPR'
export { IconPPRA } from './IconPPRA'
export { IconQuestion } from './IconQuestion'
export { IconReturn } from './IconReturn'
export { IconRisks } from './IconRisks'
export { IconRotate } from './IconRotate'
export { IconSafetyGlasses } from './IconSafetyGlasses'
export { IconScheduleDashboard } from './IconScheduleDashboard'
export { IconScheduleDashboardHour } from './IconScheduleDashboardHour'
export { IconScheduleHour } from './IconScheduleHour'
export { IconServiceOrder } from './IconServiceOrder'
export { IconServiceRequest } from './IconServiceRequest'
export { IconSmallCompany } from './IconSmallCompany'
export { IconSpeciality } from './IconSpeciality'
export { IconSuperCompany } from './IconSuperCompany'
export { IconSupplier } from './IconSupplier'
export { IconThirdParty } from './IconThirdParty'
export { IconTree } from './IconTree'
export { IconTriangleDown } from './IconTriangleDown'
export { IconTriangleUp } from './IconTriangleUp'
export { IconVest } from './IconVest'
export { IconWarehouse } from './IconWarehouse'
export { IconWarningSpecific } from './IconWarningSpecific'
export { IconWeight } from './IconWeight'
export { IconWeldingMask } from './IconWeldingMask'
export { IconWindows } from './IconWindows'
export { IconWorkPermission } from './IconWorkPermission'
export { IconYellowPending } from './IconYellowPending'
