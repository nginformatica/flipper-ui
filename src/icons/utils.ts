import { theme } from '@/theme'

const { gray, primary } = theme.colors

export interface IIconProps {
    color?: string
    card?: boolean
    white?: boolean
    active?: boolean
    viewBox?: string
    disabled?: boolean
    selected?: boolean
    height?: number | string
    width?: number | string
    isTransparent?: boolean
}

export const getIconColor = ({ active = false, card = false }: IIconProps) => {
    if (card) {
        return gray[900]
    }

    if (active) {
        return primary.main
    }

    return gray[400]
}
