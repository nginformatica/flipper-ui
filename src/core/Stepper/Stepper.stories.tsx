import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import Stepper from '.'
import { Delete } from '../../icons'

export default {
    title: 'Stepper',
    component: Stepper
} as ComponentMeta<typeof Stepper>

const Template: ComponentStory<typeof Stepper> = args => <Stepper {...args} />

export const Default = Template.bind({})
Default.args = {
    active: 2,
    steps: ['Name', 'Email', 'Password', 'Photo', 'Be happy!'],
    children: 'Stepper'
}

export const WithIcon = Template.bind({})
WithIcon.args = {
    bottomLabel: true,
    active: 1,
    steps: [
        { label: 'Name', icon: <Delete /> },
        { label: 'Name', icon: <Delete /> },
        { label: 'Name', icon: <Delete /> }
    ],
    children: 'Stepper'
}
