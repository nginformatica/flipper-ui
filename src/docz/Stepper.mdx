---
name: Stepper
route: /stepper
---

import Stepper from '../core/Stepper'
import { Delete } from '../icons'
import { Playground } from 'docz'

# Stepper

## Demo
<Playground>
    <Stepper
        active={ 2 }
        steps={ ['Name', 'Email', 'Password', 'Photo', 'Be happy!'] }
    />
    <Stepper
        bottomLabel
        active={ 2 }
        steps={ [{ label: 'Name', icon: <Delete /> }] }
    />
</Playground>
