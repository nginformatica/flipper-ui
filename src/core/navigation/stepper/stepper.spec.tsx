import * as React from 'react'
import { render, screen } from '@testing-library/react'
import { Delete } from '@/icons'
import Stepper from '.'

describe('Stepper', () => {
    it('should render', () => {
        render(
            <Stepper
                active={2}
                steps={['Name', 'Email', 'Password', 'Photo', 'Be happy!']}
            />
        )

        const container = screen.getByRole('stepper-container')
        const stepOne = screen.getByText('Name')
        const stepTwo = screen.getByText('Email')
        const stepThree = screen.getByText('Password')
        const stepFour = screen.getByText('Photo')
        const stepFive = screen.getByText('Be happy!')

        expect(container).toBeTruthy()
        expect(stepOne).toBeTruthy()
        expect(stepTwo).toBeTruthy()
        expect(stepThree).toBeTruthy()
        expect(stepFour).toBeTruthy()
        expect(stepFive).toBeTruthy()
    })

    it('should render with icon', async () => {
        render(
            <Stepper
                bottomLabel
                active={1}
                steps={[
                    {
                        label: 'Name',
                        icon: <Delete data-testid='step-icon1' />
                    },
                    {
                        label: 'Email',
                        icon: () => <Delete data-testid='step-icon2' />
                    },
                    {
                        label: 'Password',
                        icon: <Delete data-testid='step-icon3' />
                    }
                ]}
            />
        )

        const container = screen.getByRole('stepper-container')
        const stepOne = screen.getByText('Name')
        const stepOneIcon = screen.getByTestId('step-icon1')
        const stepTwo = screen.getByText('Email')
        const stepTwoIcon = screen.getByTestId('step-icon2')
        const stepThree = screen.getByText('Password')
        const stepThreeIcon = screen.getByTestId('step-icon3')

        expect(container).toBeTruthy()
        expect(stepOne).toBeTruthy()
        expect(stepOneIcon).toBeTruthy()
        expect(stepTwo).toBeTruthy()
        expect(stepTwoIcon).toBeTruthy()
        expect(stepThree).toBeTruthy()
        expect(stepThreeIcon).toBeTruthy()
        expect(stepOne.classList).toContain('MuiStepLabel-completed')
        expect(stepThree.classList).not.toContain('MuiStepLabel-completed')
    })

    it('should render with no active', async () => {
        render(
            <Stepper
                bottomLabel
                steps={[
                    {
                        label: 'Name',
                        icon: <Delete data-testid='step-icon' />
                    }
                ]}
            />
        )

        const container = screen.getByRole('stepper-container')
        const stepOne = screen.getByText('Name')
        const stepOneIcon = screen.getByTestId('step-icon')

        expect(container).toBeTruthy()
        expect(stepOne).toBeTruthy()
        expect(stepOneIcon).toBeTruthy()
        expect(stepOne.classList).not.toContain('MuiStepLabel-completed')
    })
})
