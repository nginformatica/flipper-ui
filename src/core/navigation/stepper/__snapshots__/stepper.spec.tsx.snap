// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiStepper-root MuiStepper-horizontal MuiStepper-alternativeLabel MuiPaper-elevation0"
    role="stepper-container"
  >
    <div
      class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel MuiStep-completed"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel"
      >
        <span
          class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
        >
          <svg
            active="true"
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
            data-testid="step-icon1"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
            />
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
          >
            Name
          </span>
        </span>
      </span>
    </div>
    <div
      class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel"
    >
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel MuiStepConnector-active"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel"
      >
        <span
          class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="step-icon2"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
            />
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
          >
            Email
          </span>
        </span>
      </span>
    </div>
    <div
      class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel"
    >
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
        />
      </div>
      <span
        class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel"
      >
        <span
          class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
        >
          <svg
            active="false"
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-1db085k-MuiSvgIcon-root"
            data-testid="step-icon3"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
            />
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiTypography-body2 MuiTypography-displayBlock"
          >
            Password
          </span>
        </span>
      </span>
    </div>
  </div>
</div>
`;
